cli/pack/common_test.go

```diff
@@ -19,30 +21,30 @@ func TestGetPackageFullname(t *testing.T) {
 
 	// w/o suffix
 	ctx.Project.Name = "myapp"
-	ctx.Pack.VersionRelease = "1.2.3-4"
+	ctx.Pack.VersionRelease = "1.2.3.g480c55b67"
```

Comment:
(Please, take me right: I'm still mostly out of the context of the project. So all my questions could be just because I don't dive into. And, please, don't take me as a source of truth about RPM / Deb packages: I'm not sure about all statements below.)

To be honest, I don't understand this change. I see that `normalizeVersion()` concatenates `Version` and `Release` fields. Okay. The `Version` is in the X.Y.Z form. Okay. The `Release` is either `<count>-<hash>` or `<count>` or `<hash>` or `0` (???). And there I have several questions.

1. The unit test check one case (`Release` is `<count>`), but now it checks another one (`Release` is `<hash>`). It would be easier to understand how your patch changes the behaviour if the inputs of the test case would remain unchanged. Or if you would add more test cases (and don't remove old ones) before the actual change of the code (in a separate commit).
2. As I see `VersionRelease` is formed as `<Version>-<Release>`. Not `<Version>.<Release>`. The test case is incorrect (I mean, it checks impossible situation), right?
3. I don't understand **when** `Release` is just `<count>`, when it is `<hash>`, when it is `<count>-<hash>` and when it is `0`.
4. Why the git hash is in the tarball name? That's quite strange, because a tarball source URL should be predictable by a version, if someone plan to download tarballs automatically. It is quite usual for, say, Gentoo ebuilds.
5. It also looks unusual to see the git hash in the RPM package name (however I don't know for sure). AFAIR, there is some new syntax for adding hash information into a package name, but it is modern and so rarely used now. And it is different (if I remember right, it uses a caret symbol and has its own rules).
6. AFAIR, it is usual for Debian to include a git hash into a 'preview package' or something like that. However (again, if I remember right) a release should not contain it. Whether we handle both situations correctly and it is obvious for users how to build packages in the right way?

Reply:
I can't answer the questions related to the function ``normalizeVersion``

1. Fixed
2. Right

Reply:
Now hash is stripped from all artifacts and version names are based on strict conventions. See reworked version